### PR TITLE
Disable write endpoints from configuration (fixes #26)

### LINKED ISSUE #26: Allow write operations only to certain paths
For two scenarios, it would be nice to be able to keep Syncto running for read operations, but refuse to relay write operations altogether:
- if we (initially) launch FxOS Sync only for retrieving data from the cloud, not for uploading it, it would be a nice safety guard to disable writing-to-the-cloud in the Syncto proxy.
- if we discover a problem with devices that are in use, that corrupts people's data in the cloud, we might want to disable writing-to-the-cloud at the Syncto proxy as a damage-limiting measure.

But even in normal operation, writes should only happen to the 'history', 'bookmarks' and 'passwords' collections, never for 'meta' and 'crypto'. There, read-only is enough. How hard would it be to make such a safe-guard build-time configurable?

### PR DESCRIPTION
**Pull request by leplatrem**
@Natim r?

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -133,6 +147,7 @@ But the set of settings mentionned below might deserve some review or adjustment
     cliquet.cache_pool_maxconn = 50
     fxa-oauth.cache_ttl_seconds = 3600
 
+
```
Comment (leplatrem) [inline docs/installation.rst]: useless blank line

File: syncto/views/record.py [hunk 2]
```diff
@@ -72,6 +81,14 @@ def record_put(request):
...
 def record_delete(request):
     collection_name = request.matchdict['collection_name']
+
+    # Check that endpoint is not disabled from configuration.
+    settings = request.registry.settings
+    setting_key = 'syncto.record_%s_delete_enabled' % collection_name
+    enabled = settings.get(setting_key, True)
+    if not enabled:
+        raise httpexceptions.HTTPMethodNotAllowed()
+
```
Comment (leplatrem) [inline syncto/views/record.py:97]: DRY ?

File: syncto/views/record.py [hunk 3]
```diff
@@ -49,6 +50,14 @@ def record_get(request):
 @record.put(permission=NO_PERMISSION_REQUIRED, schema=PayloadSchema)
 def record_put(request):
     collection_name = request.matchdict['collection_name']
+
+    # Check that endpoint is not disabled from configuration.
+    settings = request.registry.settings
+    setting_key = 'syncto.record_%s_put_enabled' % collection_name
+    enabled = settings.get(setting_key, True)
+    if not enabled:
+        raise httpexceptions.HTTPMethodNotAllowed()
```
Comment (leplatrem) [inline syncto/views/record.py]: Add custom message

**Comment by leplatrem** [comment]
Tests pass locally with PR on cliquet https://github.com/mozilla-services/cliquet/pull/471

File: docs/installation.rst [hunk 4]
```diff
@@ -120,8 +120,21 @@ Recommended settings
 
 Most default setting values in the application code base are suitable for production.
 
-But the set of settings mentionned below might deserve some review or adjustments:
+However, **for safety reasons**, it might appropriate to disable write operations
```
Comment (Natim) [inline docs/installation.rst]: it might be

File: docs/installation.rst [hunk 5]
```diff
@@ -120,8 +120,21 @@ Recommended settings
...
+to remote Sync collections:
 
+.. code-block :: ini
+
+    syncto.record_tabs_put_enabled = false
```
Comment (Natim) [inline docs/installation.rst:128]: I would add a `syncto.record_tabs.read-only = true` shortcut.

File: syncto/tests/test_functional.py [hunk 6]
```diff
@@ -293,3 +286,33 @@ def test_put_return_a_404_in_case_of_unknown_resource(self):
...
+        self.app.put_json(RECORD_URL, RECORD_EXAMPLE,
+                          headers=self.headers, status=405)
+
+    def test_record_delete_is_disabled(self):
+        self.app.delete(RECORD_URL, headers=self.headers, status=405)
+
+    def test_message_provides_details(self):
+        r = self.app.put_json(RECORD_URL, RECORD_EXAMPLE,
+                              headers=self.headers, status=405)
+        expected = 'Endpoint disabled for this collection in configuration.'
```
Comment (Natim) [inline syncto/tests/test_functional.py]: Please use the [FormattedErrorMixin](https://github.com/mozilla-services/cliquet/blob/master/cliquet/tests/test_views_errors.py#L53-L55)

File: docs/installation.rst (hunk 5 above)
Comment (leplatrem) [inline docs/installation.rst:128]: I wanted to respect the resource settings to disable endpoint by method...

**Comment by Natim** [comment]
Rebased.

**Pull request merged**